### ISSUE TITLE
OFI MTL - HPE CXI provider problem with synchronous sends

### ISSUE THREAD

**Issue by hppritcha** [open-mpi/ompi#10210]
Running IBM regression test suite using OMPI master and the HPE provided libfabric shows failures that aren't observed using the the OFI MTL with rxm;verbs on a IB cluster:

```
FAIL: comm_idup
FAIL: comm_idup_usempif08
FAIL: comm_idup_usempi
FAIL: comm_idup_mpifh
```

The common feature of these tests is the use of MPI_Ssend.  Converting the MPI_Ssend's to MPI_Send causes the tests to pass.

**Comment by hppritcha** [comment]
this was determined to be an issue with the vendor provider.   patch supplied to vendor that fixes the problem.
related to https://github.com/ofiwg/libfabric/issues/7652

**Comment by jsquyres** [comment]
@hppritcha This was merged upstream.  What do you want to do with this issue?

**Comment by hppritcha** [comment]
HI Jeff, what was merged upstream?  Something in OFI libfabric?

**Comment by jsquyres** [comment]
@hppritcha My mistake -- I looked at https://github.com/ofiwg/libfabric/issues/7652 too quickly.  Disregard my above comment...

**Comment by benkirk** [comment]
I'm curious if you ever found resolution to this problem?

FWIW, I've seen this on our development Cray EX and reproduced most recently (today) with OpenMPI-4.1.5 and `/opt/cray/libfabric/1.15.2.0`

**Comment by hppritcha** [comment]
No.  The problem is in the CXI provider as noted in the issue.  Unfortunately the vendor has elected to date to not. upstream the CXI source code to the community libfabric project so I can't open a PR with a fix.

**Comment by hppritcha** [comment]
oh sorry Howard forgot he actually opened a ticket with the vendor and included the supplied patch.

**Comment by hppritcha** [comment]
Okay this is so annoying.  By reverse engineering the bug in the CXI fi_tsendmsg error checking code I may have a fix.  Turns out one of my employer's important applications makes use of synchronous sends.  I'll open a PR after some more testing.  I don't plan to patch 4.1.x branch so you need to move to the 5.0.0 pre-release candidates @benkirk

**Comment by hppritcha** [comment]
closed via #11718 and #11720
